Lib/test/test_dataclasses.py

```diff
@@ -2150,6 +2150,25 @@ def __init__(self, x):
                 self.x = 2 * x
         self.assertEqual(C(5).x, 10)
 
+    def test_inherit_from_protocol(self):
+        # See bpo-45081.
```

Comment:
To save future generations some time. Please change this to
```suggestion
        # Dataclasses inheriting from protocol should preserve their own `__init__`.
        # See bpo-45081.
```